tools/generate-code.py

```diff
@@ -116,9 +116,9 @@ def generate_fromwire_field(field, allfields, lang):
                       .format(fname=field.name, size=sizestr), file=ofile)
         else:
             if lang == 'js':
-                print('    v = [];\n'
+                print('    v = "";\n'
                       '    for (let i = 0; {limit}; i++) {{\n'
-                      '        v.push(fromwire_{ftype}(buffer));\n'
+                      '        v+=(fromwire_{ftype}(buffer));\n'
                       '    }}\n'
```

Comment:
I think this change is wrong?  If we're not dealing with utf8, we really do want this to be a [], not a string (e.g. if it's an array of chain_hash).